modules/vmseries/variables.tf

```diff
@@ -1,219 +1,232 @@
-  description = "VM-series SKU - list available with `az vm image list -o table --all --publisher paloaltonetworks`"
-  default     = "byol"
-  type        = string
-}
-
-variable "img_version" {
-  description = "VM-series PAN-OS version - list available for a default `img_offer` with `az vm image list -o table --publisher paloaltonetworks --offer vmseries-flex --all`"
-  default     = "10.1.0"
-  type        = string
-}
-
-variable "tags" {
-  description = "A map of tags to be associated with the resources created."
-  default     = {}
-  type        = map(any)
-}
-
-variable "identity_type" {
-  description = "See the [provider documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine#identity_type)."
-  default     = "SystemAssigned"
-  type        = string
-}
-
-variable "identity_ids" {
-  description = "See the [provider documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine#identity_ids)."
-  default     = null
-  type        = list(string)
-}
-
-variable "accelerated_networking" {
-  description = "Enable Azure accelerated networking (SR-IOV) for all network interfaces except the primary one (it is the PAN-OS management interface, which [does not support](https://docs.paloaltonetworks.com/pan-os/9-0/pan-os-new-features/virtualization-features/support-for-azure-accelerated-networking-sriov) acceleration)."
-  default     = true
-  type        = bool
-}
-
-variable "bootstrap_options" {
-  description = <<-EOF
-  Bootstrap options to pass to VM-Series instance.
-
-  Proper syntax is a string of semicolon separated properties.
-  Example:
-    bootstrap_options = "type=dhcp-client;panorama-server=1.2.3.4"
-
-  A list of available properties: storage-account, access-key, file-share, share-directory, type, ip-address, default-gateway, netmask, ipv6-address, ipv6-default-gateway, hostname, panorama-server, panorama-server-2, tplname, dgname, dns-primary, dns-secondary, vm-auth-key, op-command-modes, op-cmd-dpdk-pkt-io, plugin-op-commands, dhcp-send-hostname, dhcp-send-client-id, dhcp-accept-server-hostname, dhcp-accept-server-domain, auth-key, vm-series-auto-registration-pin-value, vm-series-auto-registration-pin-id.
-
-  For more details on bootstrapping see documentation: https://docs.paloaltonetworks.com/vm-series/10-2/vm-series-deployment/bootstrap-the-vm-series-firewall/create-the-init-cfgtxt-file/init-cfgtxt-file-components
-  EOF
-  default     = ""
-  type        = string
-  sensitive   = true
-}
-
-variable "diagnostics_storage_uri" {
-  description = "The storage account's blob endpoint to hold diagnostic files."
-  default     = null
-  type        = string
+  type = list(object({
+    name                     = string
+    subnet_id                = string
+    create_public_ip         = optional(bool, false)
+    public_ip_name           = optional(string)
+    public_ip_resource_group = optional(string)
```

Comment:
Is it name of the resource group, not key , yes ? Then maybe we should use name  ` public_ip_resource_group_name` instead of ` public_ip_resource_group` ?

Reply:
well you are probably right, the problem is that we already use this convention in other modules, like [`load balancer`](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/a63213f80532ba637c7d785f53d70c642a0ee59a/modules/loadbalancer/variables.tf#L72)

so if we decide to change it, it has to be everywhere

and besides, if a property holds a key, it should be prefixed with a `_key`, like [here for example](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/ff2b20e5f0770267b3ab6bccd8b65a682f26fc43/examples/vmseries/variables.tf#L376)

Reply:
this was fixed in all examples and modules, from now on all properties that hold a name are suffixed with `_name`